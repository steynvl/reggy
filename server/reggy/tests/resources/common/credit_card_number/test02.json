{
  "type": "Credit card number",
  "information": {
    "visa": false,
    "dinersClub": false,
    "masterCard": true,
    "discover": false,
    "americanExpress": false,
    "jcb": false,
    "spacesAndDashes": "Spaces and dashes to group digits"
  },
  "generalRegexInfo": {
    "startRegexMatchAt": "Anywhere",
    "endRegexMatchAt": "Anywhere",
    "regexTarget": "Java"
  },
  "generateMethod": "commonUseCases"
}
