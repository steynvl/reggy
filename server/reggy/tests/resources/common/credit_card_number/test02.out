Java
Pattern regex = Pattern.compile("(?:5[1-5]\\d{2}|2720|27[01][0-9]|2[3-6][0-9]{2}|22[3-9][0-9]|222[1-9])[ -]*\\d{4}[ -]*\\d{4}[ -]*\\d{4}");
(?:5[1-5]\\d{2}|2720|27[01][0-9]|2[3-6][0-9]{2}|22[3-9][0-9]|222[1-9])[ -]*\\d{4}[ -]*\\d{4}[ -]*\\d{4}
Perl
my $regex = /(?:5[1-5]\d{2}|2720|27[01][0-9]|2[3-6][0-9]{2}|22[3-9][0-9]|222[1-9])[ -]*\d{4}[ -]*\d{4}[ -]*\d{4}/;
(?:5[1-5]\d{2}|2720|27[01][0-9]|2[3-6][0-9]{2}|22[3-9][0-9]|222[1-9])[ -]*\d{4}[ -]*\d{4}[ -]*\d{4}
POSIX
(?:5[1-5]\d{2}|2720|27[01][0-9]|2[3-6][0-9]{2}|22[3-9][0-9]|222[1-9])[ -]*\d{4}[ -]*\d{4}[ -]*\d{4}
(?:5[1-5]\d{2}|2720|27[01][0-9]|2[3-6][0-9]{2}|22[3-9][0-9]|222[1-9])[ -]*\d{4}[ -]*\d{4}[ -]*\d{4}
Python
regex = re.compile(r'(?:5[1-5]\d{2}|2720|27[01][0-9]|2[3-6][0-9]{2}|22[3-9][0-9]|222[1-9])[ -]*\d{4}[ -]*\d{4}[ -]*\d{4}')
(?:5[1-5]\d{2}|2720|27[01][0-9]|2[3-6][0-9]{2}|22[3-9][0-9]|222[1-9])[ -]*\d{4}[ -]*\d{4}[ -]*\d{4}
JavaScript
const regex = /(?:5[1-5]\d{2}|2720|27[01][0-9]|2[3-6][0-9]{2}|22[3-9][0-9]|222[1-9])[ -]*\d{4}[ -]*\d{4}[ -]*\d{4}/;
(?:5[1-5]\d{2}|2720|27[01][0-9]|2[3-6][0-9]{2}|22[3-9][0-9]|222[1-9])[ -]*\d{4}[ -]*\d{4}[ -]*\d{4}
PHP
$regex = '/(?:5[1-5]\d{2}|2720|27[01][0-9]|2[3-6][0-9]{2}|22[3-9][0-9]|222[1-9])[ -]*\d{4}[ -]*\d{4}[ -]*\d{4}/';
(?:5[1-5]\d{2}|2720|27[01][0-9]|2[3-6][0-9]{2}|22[3-9][0-9]|222[1-9])[ -]*\d{4}[ -]*\d{4}[ -]*\d{4}
Golang
regex, _ := regexp.Compile("(?:5[1-5]\\d{2}|2720|27[01][0-9]|2[3-6][0-9]{2}|22[3-9][0-9]|222[1-9])[ -]*\\d{4}[ -]*\\d{4}[ -]*\\d{4}")
(?:5[1-5]\\d{2}|2720|27[01][0-9]|2[3-6][0-9]{2}|22[3-9][0-9]|222[1-9])[ -]*\\d{4}[ -]*\\d{4}[ -]*\\d{4}
Rust
let re = Regex::new(r"(?:5[1-5]\d{2}|2720|27[01][0-9]|2[3-6][0-9]{2}|22[3-9][0-9]|222[1-9])[ -]*\d{4}[ -]*\d{4}[ -]*\d{4}").unwrap();
(?:5[1-5]\d{2}|2720|27[01][0-9]|2[3-6][0-9]{2}|22[3-9][0-9]|222[1-9])[ -]*\d{4}[ -]*\d{4}[ -]*\d{4}
C#
var re = new Regex(@"(?:5[1-5]\d{2}|2720|27[01][0-9]|2[3-6][0-9]{2}|22[3-9][0-9]|222[1-9])[ -]*\d{4}[ -]*\d{4}[ -]*\d{4}");
(?:5[1-5]\d{2}|2720|27[01][0-9]|2[3-6][0-9]{2}|22[3-9][0-9]|222[1-9])[ -]*\d{4}[ -]*\d{4}[ -]*\d{4}
Scala
val re = "(?:5[1-5]\\d{2}|2720|27[01][0-9]|2[3-6][0-9]{2}|22[3-9][0-9]|222[1-9])[ -]*\\d{4}[ -]*\\d{4}[ -]*\\d{4}".r
(?:5[1-5]\\d{2}|2720|27[01][0-9]|2[3-6][0-9]{2}|22[3-9][0-9]|222[1-9])[ -]*\\d{4}[ -]*\\d{4}[ -]*\\d{4}
Kotlin
val regex = Regex("(?:5[1-5]\\d{2}|2720|27[01][0-9]|2[3-6][0-9]{2}|22[3-9][0-9]|222[1-9])[ -]*\\d{4}[ -]*\\d{4}[ -]*\\d{4}")
(?:5[1-5]\\d{2}|2720|27[01][0-9]|2[3-6][0-9]{2}|22[3-9][0-9]|222[1-9])[ -]*\\d{4}[ -]*\\d{4}[ -]*\\d{4}